### PR TITLE
Add CLI tests for duplicate CV environments dedup

### PR DESCRIPTION
**Pull request by jnagare-redhat**
## Summary
- Adds `test_positive_update_ak_with_duplicate_cv_envs` in `tests/foreman/cli/test_activationkey.py` to verify that updating an activation key with duplicate content-view-environments deduplicates them silently
- Adds `test_positive_host_update_with_duplicate_cv_envs` in `tests/foreman/cli/test_contentview.py` to verify that updating a host with duplicate content-view-environments deduplicates them silently
- Both tests verify SAT-42052 (upstream fix: https://github.com/Katello/katello/pull/11813)

## Test plan
- [x] Both tests collected successfully
- [x] Both tests passed against Satellite with the fix applied

### PRT
```
pytest tests/foreman/cli/test_activationkey.py::test_positive_update_ak_with_duplicate_cv_envs tests/foreman/cli/test_contentview.py::TestContentView::test_positive_host_update_with_duplicate_cv_envs
```

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Add CLI tests ensuring duplicate content view environments are silently deduplicated when updating activation keys and hosts.

Tests:
- Add high-importance regression test for activation key updates with duplicate content view environments.
- Add high-importance regression test for host updates with duplicate content view environments.

### CODE REVIEW

**Comment by jnagare-redhat** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_activationkey.py::test_positive_update_ak_with_duplicate_cv_envs tests/foreman/cli/test_contentview.py::TestContentView::test_positive_host_update_with_duplicate_cv_envs
Katello:
    katello: 11813

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 16321
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_activationkey.py::test_positive_update_ak_with_duplicate_cv_envs tests/foreman/cli/test_contentview.py::TestContentView::test_positive_host_update_with_duplicate_cv_envs --external-logging
Test Result : ================= 2 passed, 31 warnings in 2299.99s (0:38:19) ==================
```

**Review by synkd — APPROVED** [review]

File: tests/foreman/cli/test_activationkey.py [hunk 1]
```diff
@@ -1598,6 +1598,55 @@ def test_positive_multi_cv_info_and_remove_all_cv_envs(
...
+        2. Create an activation key
+        3. Update the activation key with duplicate content view environments
+           (e.g., cv1/lce1,cv2/lce2,cv1/lce1)
+
+    :expectedresults: The activation key is updated successfully with only the unique
+        content view environments, duplicates are silently ignored.
+
+    :Verifies: SAT-42052
+
+    :CaseImportance: High
```
Comment (LadislavVasina1) [inline tests/foreman/cli/test_activationkey.py]: @jnagare-redhat can we please drop this, we are not using this to collect tests, or give us some kind of sorting
```suggestion
```

File: tests/foreman/cli/test_contentview.py [hunk 2]
```diff
@@ -3231,6 +3231,69 @@ def test_negative_inc_update_with_existing_errata(self, module_org, module_targe
...
+            3. Create a fake host associated with the first content view environment
+            4. Update the host with duplicate content view environments
+               (e.g., cv1/lce1,cv2/lce2,cv1/lce1)
+
+        :expectedresults: The host is updated successfully with only the unique
+            content view environments, duplicates are silently deduplicated.
+
+        :Verifies: SAT-42052
+
+        :CaseImportance: High
```
Comment (LadislavVasina1) [inline tests/foreman/cli/test_contentview.py]:
```suggestion
```
same as above

**Comment by jnagare-redhat** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/cli/test_activationkey.py::test_positive_update_ak_with_duplicate_cv_envs tests/foreman/cli/test_contentview.py::TestContentView::test_positive_host_update_with_duplicate_cv_envs
Katello:
    katello: 11813
```

**Review by LadislavVasina1 — APPROVED** [review]

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 16326
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_activationkey.py::test_positive_update_ak_with_duplicate_cv_envs tests/foreman/cli/test_contentview.py::TestContentView::test_positive_host_update_with_duplicate_cv_envs --external-logging
Test Result : ================= 2 passed, 30 warnings in 1406.26s (0:23:26) ==================
```

**Review by vijaysawant — APPROVED** [review]

**Review by vijaysawant — COMMENTED** [review]
Lets use `settings_update` fixture to update the value of `allow_multiple_content_views` prior to test execution.

**Comment by jnagare-redhat** [comment]
> Lets use `settings_update` fixture to update the value of `allow_multiple_content_views` prior to test execution.

Its default to true. no need to separately update.

**Comment by jnagare-redhat** [comment]
Katello chanes are not merge yet. Hold this PR till now.

**Review by ColeHiggins2 — APPROVED** [review]

**Comment by jnagare-redhat** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_activationkey.py::test_positive_update_ak_with_duplicate_cv_envs tests/foreman/cli/test_contentview.py::TestContentView::test_positive_host_update_with_duplicate_cv_envs

**Comment by jnagare-redhat** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_activationkey.py::test_positive_update_ak_with_duplicate_cv_envs tests/foreman/cli/test_contentview.py::TestContentView::test_positive_host_update_with_duplicate_cv_envs

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 16444
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_activationkey.py::test_positive_update_ak_with_duplicate_cv_envs tests/foreman/cli/test_contentview.py::TestContentView::test_positive_host_update_with_duplicate_cv_envs --external-logging
Test Result : ================= 2 passed, 30 warnings in 1114.83s (0:18:34) ==================
```
